spec/queue.js

```diff
@@ -0,0 +1,117 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import Queue from '../src/queue'
+import Node from '../src/node'
+
+chai.use(chaiChange)
+
+describe('Queue', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(Queue).to.be.a('function')
+  })
+
+  context('enqueue()', () => {
+    it('adds an element to the back of the queue.', () => {
+      const myQueue = new Queue()
+
+      expect(() => myQueue.enqueue('foo'))
+        .to.alter(() => myQueue.length(), { from: 0, to: 1 })
+    })
+  })
+
+  context('dequeue()', () => {
+
+    context('when the queue is not empty', () => {
+      it('removes and returns the front element', () => {
+        const myQueue = new Queue()
+
+        myQueue.enqueue('foo')
+        myQueue.enqueue('bar')
+        expect(myQueue.dequeue()).to.be.an.instanceof(Node)
+      })
+    })
+
+    context('when the queue is empty', () => {
+      it('returns null', () => {
+        const myQueue = new Queue()
+
+        expect(myQueue.dequeue()).to.equal(null)
+      })
+    })
+  })
+
+
+  context('front()', () => {
+    context('when the queue is not empty', () => {
+      it('returns the front element', () => {
+        const myQueue = new Queue()
+
+        myQueue.enqueue('foo')
+        myQueue.enqueue('bar')
```

Comment:
You seem to be creating similar queues to run your tests. Consider looking into the beforeEach function to reduce the code you have to manually write.